### DISCUSSION TITLE
Group avatar  - file format not supported issue

### DISCUSSION THREAD

**Discussion by trafford-tbe** [python-gitlab/python-gitlab#2751] [Q&A] [answered]
## Description of the problem, including code/CLI snippet
I'm trying to add avatar to group using following command:

`gitlab -v group create --name "Test 1" --path "test1" --parent-id 21 --avatar "/tmp/logo_lab.png"`

where:
1. Group creation without `--avatar` option works correctly.
2. Path to .png is absulute path to file with correct read permissions.
3. The same file added using web-gui dispalys correctly.

## Expected Behavior
Avatar added to group.

## Actual Behavior
Message returned:
`Impossible to create object (400: Failed to save group {:avatar=>["file format is not supported. Please try one of the following supported formats: image/png, image/jpeg, image/gif, image/bmp, image/tiff, image/vnd.microsoft.icon"]})
`

## Specifications

  - python-gitlab version: 4.2.0
  - API version you are using (v3/v4): v4
  - Gitlab server version (or gitlab.com): v16.6.2-ee

**Comment by JohnVillalovos** [comment]
I think you will need to do something like: https://python-gitlab.readthedocs.io/en/main/cli-usage.html#reading-values-from-files

`gitlab -v group create --name "Test 1" --path "test1" --parent-id 21 --avatar "@/tmp/logo_lab.png"`

**Comment by trafford-tbe** [comment]
Already tested with following results:
`'utf-8' codec can't decode byte 0x89 in position 0: invalid start byte`
which seems to be answer from python.

**Reply by JohnVillalovos** [reply]
Ah. The code for the `@` prefix assumes it is a text file :(

So at the moment I don't think it can be done via the CLI. With the API it can be done.

**Reply by trafford-tbe** [reply]
Yes. I made it with cURL, but tried to use one module for Ansible:
https://docs.ansible.com/ansible/latest/collections/community/general/gitlab_group_module.html
Nevertheless, thank you for confirmation, but does this mean we have an issue now?

**Reply by JohnVillalovos** [reply, marked as answer]
I have created an issue #2752 to track it. I don't have an ETA on when someone will volunteer to work on it.
